types/precison_timestamp_types.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
+
+const (
+	UnknownPrecision TimePrecision = iota
+	Seconds
+	DeciSeconds
+	CentiSeconds
+	MilliSeconds
+	EMinus4Seconds // 10^-4 of seconds
+	EMinus5Seconds // 10^-5 of seconds
+	MicroSeconds
+	EMinus7Seconds // 10^-7 of seconds
+	EMinus8Seconds // 10^-8 of seconds
+	NanoSeconds
+)
+
+func timePrecisionToProtoVal(val TimePrecision) int32 {
+	switch val {
+	case Seconds:
+		return 0
+	case DeciSeconds:
+		return 1
+	case CentiSeconds:
+		return 2
+	case MilliSeconds:
+		return 3
+	case EMinus4Seconds:
+		return 4
+	case EMinus5Seconds:
+		return 5
+	case MicroSeconds:
+		return 6
+	case EMinus7Seconds:
+		return 7
+	case EMinus8Seconds:
+		return 8
+	case NanoSeconds:
+		return 9
```

Comment:
I'm confused by this function, why not change the enum constants so that the values match these values? thus conversion would just be `int(val)` ?

You could do this by making `UnknownPrecision = -1` and then use `iota` starting with `Seconds`

Reply:
agree on improvement with ^

Reply:
fixed it, kindly recheck